tests/pytest/addons.py

```diff
@@ -31,6 +31,30 @@ def is_numpy_new_enough(version_feature_introduced):
     return parse_version(numpy.version.version) >= parse_version(version_feature_introduced)
 
 
+def is_nvidia_gpu_present():
+    try:
+        import GPUtil
+    except ImportError:  # py36 ModuleNotFoundError
+        try:
+            import gpu_dfcc
+        except ImportError:  # py36 ModuleNotFoundError
+            # who knows?
+            return False
+        else:
+            return gpu_dfcc.cudaGetDeviceCount() > 0
+    else:
+        try:
+            ngpu = len(GPUtil.getGPUs())
+        except OSError:  # py3 FileNotFoundError
+            # no `nvidia-smi`
+            return False
+        else:
+            return ngpu > 0
+
+
+hardware_nvidia_gpu = pytest.mark.skipif(is_nvidia_gpu_present() is False,
```

Comment:
Rought.

Reply:
agreed. exporting the nGPUs from `gpu_dfcc` would be better, but that needs plugin overhaul, pybind11. Or I could write my own subprocess/nvidia-smi/count_gpus fn, but I don't dare do that w/o an `nvidia-wmi` cmd to test on. hence, a hack.

Reply:
Agree that this is probably the best solution for now. I wouldn't recommend writing out our own logic here.